### ISSUE TITLE
md-autocomplete doesn't play nice with cordova-plugin-googlemaps

### ISSUE THREAD

**Issue by remcohaszing** [angular/material#8201]
I'm using `angular-material` with [cordova-plugin-googlemaps](https://github.com/mapsplugin/cordova-plugin-googlemaps).

To use this plugin, a developer has to select an element. This will then be made invisible along with all its parents and interaction with this element will be delegated with a Google Maps instance behind the webview.

Elements rendered within this element are rendered normally and can be interacted with.

I have roughly the following HTML:
```html
<body>
  <div>
    <my-map>  <!-- custom directive to which the Google Maps instance will be attached -->
      <div>  <!-- Anything inside this div will behave normally -->
        <md-autocomplete></md-autocomplete>
      </div>
    </my-map>
  </div>
</body>
```

The problem here is that `<md-autocomplete>` attaches the search results to a `<md-virtual-repeat-container>` at the end of the body element. This means any overflowing elements won't be clickable, but will send the events to the Google Maps instance instead.

The following screenshot shows what's happening:

* Clicking the upper results, e.g. `Netherlands` will work as expected.
* Clicking results at the bottom, e.g. `Netherfield, United Kingdom` will send the event to Google Maps.

![screenshot_2016-04-25_12-55-31](https://cloud.githubusercontent.com/assets/779047/14781891/bcb05682-0ae5-11e6-9f3f-a7fed5e794cb.png)

I think the best solution would be to attach the `<md-virtual-repeat-container>` to a child element of `<md-autocomplete>` instead of the body.

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.

**Comment by GuelorEmanuel** [comment]
@remcohaszing 
How did you set up the custom directive  for attaching the Google map instance. For some reason I cant click the angular material button "test" on the foreground, every interaction is being delegated to the map.

<md-content  flex>
    <div id="map_canvas">
    </div>
     <md-button class="md-raised mm-button" layout-margin ng-click="homeRegualrVM.test()">
            Test
    </md-button>
</md-content>

**Comment by remcohaszing** [comment]
@GuelorEmanuel 
Anything not in the map element will delegate user interaction to the map. (https://github.com/mapsplugin/cordova-plugin-googlemaps/wiki/Map#put-html-elements-on-the-map)

So I suppose your HTML should be rewritten to something like:

```html
<div id="map_canvas" layout="column" layout-fill layout-align="bottom" flex>
  <md-content flex>
    <md-button class="md-raised mm-button" layout-margin ng-click="homeRegularVM.test()">
      Test
    </md-button>
  </md-content>
</div>
```

Note that I also changed `homeRegualrVM` to `homeRegularVM`.

For controls that get attached to the body, e.g. side menus / dialogs, make sure to disable the map before showing them.

**Comment by GuelorEmanuel** [comment]
@remcohaszing 

Thank you so much, this helped me a lot!

**Comment by GuelorEmanuel** [comment]
@remcohaszing 
I just have one more question, are you using cordova-plugin-googlemaps for the Google Places Autocomplete service if so how?

**Comment by remcohaszing** [comment]
Yep. I'm using the Google Places API Web Service for that, because `cordova-plugin-googlemaps` doesn't support it.
